checkov/circleci_pipelines/image_referencer/provider.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+from typing import Any
+import jmespath
+
+from checkov.common.images.image_referencer import Image
+
+
+class CircleCIProvider:
+    __slots__ = ("workflow_config", "file_path")
+
+    def __init__(self, workflow_config: dict[str, Any], file_path: str):
+        self.file_path = file_path
+        self.workflow_config = workflow_config
+
+    def extract_images_from_workflow(self) -> list[Image]:
+        images: list[Image] = []
```

Comment:
maybe should be a set? Is a List in other implementations

Reply:
code-wise it should be possible. at the moment there was no need to do it, because getting a duplicated `Image` instance is rather rare and shouldn't happen. But what we don't do and this would be an improvement to only send unique image names to our platform. Are you interested to do this change in a separate PR?